### ISSUE TITLE
Add some additional syncing related information to help solve syncing issues

### ISSUE THREAD

**Issue by biblicabeebli** [Automattic/simplenote-electron#2162]
Please, be as descriptive as possible.  Issues lacking detail, or for any other reason than to request a feature, may be closed without action.

### What
***(Required)*** What is the feature you are requesting?  Add a concise description of the feature being requested.

Improve the information visible to the user and enhance the user interface when looking at note history.

In no particular order here are some ideas (I don't think any of these conflict, please comment if you disagree):

1) Indicate the source device of each change when dragging around the history bar.

2) Indicate both the authored date+time of the changes, and the sync event date+time of the changes.  Possibly highlight items where these differ dramaticaly (I'm thinking 1 counts as "dramatic").

3) (upon reflection, this one may not be in the spirit of "simple", but it might be elegant enough to count. :b )
Provide a higher-detail, vertical, clickable list view of the history events of a note.  The lote listing side panel of the app could transform into a list of events that would have a highlighted element that would scroll up and down as the user drags the bar, and clicking on an item would move the bar.  This would be easier to scan through when looking at long histories, especially when the user knows what they are searching for.  It will visually highlight potential discontinuities in the syncing process, and help users resolve those issues, especially combined with idea 1 and 2 above.

4) Highlight history events that are purely the removal of text.  These are likely quite common and/or tend to be grouped together.  This would offer an affordance that helps the user place themself in the context of the version of the text they are searching for.  It could be implemented as a color dot on the drag bar, or line coloration in a text-based listing.

4.1) (similar to 4) Indicate sync events that result in reversion of a note to an earlier state, and indicate the timestamp of that earlier state.

4.2) (similar to 4.1) Indicate an event when a user has manually taken the "Restore Note" action, and indicate or even point to the timestamp of the source state.

(is the difference between 4.1 and 4.2 clear?)

### Why
***(Required)*** Why are you requesting this feature?  Add a concise description of the problem this feature solves.

Syncing bugs are a fact of life, sometimes you have to deal with a bad sync, and sometimes you have to find a historical version of a note.

When navigating history there is some low-hanging fruit options that would help surface information a user would find helpful.

**Comment by dmsnell** [comment]
Hi @biblicabeebli 👋 

The good news is that we're currently working on some of this. We won't likely implement some of the more complicated history views and may not be able to indicate which revisions were a restore (this is due to how the underlying data model works).

What's on the docket is some indication for when a note was last synchronized, when it receives an update from another device, and if local changes have been confirmed to be synchronized on the server.

**Comment by biblicabeebli** [comment]
Awesome.

I think indicating source device is most important because that is where most (all?) syncing bugs come from.

I think that indicating "this update just deleted stuff" is the best data point to display if the available display mechanism is limited.  I've gone in to check when I _added_ something never, I've gone in to _view change over time_ exactly twice, but I go in to find something I deleted or got removed from the history due to a weird sync up to a couple times a week.
